sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java

```diff
@@ -35,102 +19,73 @@
-
-    return result;
+  ResponseEntity<String> completion() {
+    final var content = service.completion().getContent();
+    log.info("Our trusty AI answered with: {}", content);
+    return ResponseEntity.ok(content);
   }
 
   /**
    * Asynchronous stream of an OpenAI chat request
    *
    * @return the emitter that streams the assistant message response
    */
-  @SuppressWarnings("unused") // The end-to-end test doesn't use this method
   @GetMapping("/streamChatCompletion")
   @Nonnull
   public ResponseEntity<ResponseBodyEmitter> streamChatCompletion() {
-    final var prompt =
-        new OrchestrationPrompt("Can you give me the first 100 numbers of the Fibonacci sequence?");
-    final var stream = client.streamChatCompletion(prompt, config);
-
-    final var emitter = new ResponseBodyEmitter();
-
-    final Runnable consumeStream =
-        () -> {
-          try (stream) {
-            stream.forEach(
-                deltaMessage -> {
-                  log.info("Controller: {}", deltaMessage);
-                  send(emitter, deltaMessage);
-                });
-          } finally {
-            emitter.complete();
-          }
-        };
-
-    ThreadContextExecutors.getExecutor().execute(consumeStream);
-
-    // TEXT_EVENT_STREAM allows the browser to display the content as it is streamed
-    return ResponseEntity.ok().contentType(MediaType.TEXT_EVENT_STREAM).body(emitter);
+    return service.streamChatCompletion();
   }
 
   /**
    * Chat request to OpenAI through the Orchestration service with a template.
    *
    * @link <a href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/templating">SAP
    *     AI Core: Orchestration - Templating</a>
-   * @return the result object
+   * @return a ResponseEntity with the response content
    */
   @GetMapping("/template")
   @Nonnull
-  OrchestrationChatResponse template() {
-    final var template =
-        Message.user("Reply with 'Orchestration Service is working!' in {{?language}}");
-    final var templatingConfig = Template.create().template(List.of(template.createChatMessage()));
-    final var configWithTemplate = config.withTemplateConfig(templatingConfig);
-
-    final var inputParams = Map.of("language", "German");
-    final var prompt = new OrchestrationPrompt(inputParams);
+  ResponseEntity<String> template() {
+    return ResponseEntity.ok(service.template().getContent());
+  }
 
-    return client.chatCompletion(prompt, configWithTemplate);
+  /**
+   * Chat request to OpenAI through the Orchestration service with a template and full response.
+   *
+   * @link <a href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/templating">SAP
+   *     AI Core: Orchestration - Templating</a>
+   * @return a ResponseEntity with the full response
+   */
+  @GetMapping("/template/full")
+  @Nonnull
+  ResponseEntity<String> templateFull() {
+    return ResponseEntity.ok(service.template().toString());
```

Comment:
Should I add such a view for the full response object for each of the endpoints? If yes, should all of them also be shown in the `index.html`?

Reply:
`toString()` will not be helpful I'm afraid. @TillK17 is preparing a JSON syntax viewer in the upcoming frontend update. And this method does not produce proper JSON.

Other options:
* Use custom `ObjectMapper` that only looks for fields upon serialization
  * serialize to and return String
  * serialize to and return JsonObject
* Use GSON serializer as alternative (not recommended)

Maybe we can draft more ideas.

Reply:
> Use custom `ObjectMapper` that only looks for fields upon serialization

This is what I mean:
```java 
var mapper = new ObjectMapper().setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
mapper.writeValueAsString(service.template()): // String
```

Not tested. You can use static imports to make it fit into one line.